fluss-common/src/main/java/org/apache/fluss/row/InternalArray.java

```diff
@@ -184,6 +185,29 @@ static ElementGetter createDeepElementGetter(DataType fieldType) {
                             return new GenericArray(objs);
                         };
                 break;
+            case MAP:
+                DataType keyType = ((org.apache.fluss.types.MapType) fieldType).getKeyType();
+                DataType valueType = ((org.apache.fluss.types.MapType) fieldType).getValueType();
+                ElementGetter keyGetter = createDeepElementGetter(keyType);
+                ElementGetter valueGetter = createDeepElementGetter(valueType);
+                elementGetter =
+                        (array, pos) -> {
+                            InternalMap inner = array.getMap(pos);
+                            InternalArray keys = inner.keyArray();
+                            InternalArray values = inner.valueArray();
+                            Object[] keyObjs = new Object[keys.size()];
+                            Object[] valueObjs = new Object[values.size()];
+                            for (int i = 0; i < keys.size(); i++) {
+                                keyObjs[i] = keyGetter.getElementOrNull(keys, i);
+                                valueObjs[i] = valueGetter.getElementOrNull(values, i);
```

Comment:
we don't need the temporary `keyObjs` and `valueObjs`, directly put them into the map, this can reduce small object overhead.